ResearchKit/Common/ORKTaskViewController.m

```diff
@@ -1162,19 +1171,10 @@ - (IBAction)flipToPreviousPageFrom:(ORKStepViewController *)fromController {
     ORKStepViewController *stepViewController = nil;
     
     if ([self shouldPresentStep:step]) {
-        ORKStep *currentStep = _currentStepViewController.step;
-        NSString *itemId = currentStep.identifier;
-        
         stepViewController = [self viewControllerForStep:step];
-        
         if (stepViewController) {
-            // Remove the identifier from the list
-            assert([itemId isEqualToString:_managedStepIdentifiers.lastObject]);
-            [_managedStepIdentifiers removeLastObject];
```

Comment:
This is managed as the browser's back stack, remove current item from stack when user hit `back button`.